packages/reactive-element/src/decorators/query.ts

```diff
@@ -59,50 +60,63 @@ export type QueryDecorator = {
 export function query(selector: string, cache?: boolean): QueryDecorator {
   return (<C extends Interface<ReactiveElement>, V extends Element>(
     protoOrTarget: ClassAccessorDecoratorTarget<C, V>,
-    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>
+    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>,
+    descriptor?: PropertyDescriptor
   ) => {
     const doQuery = (el: Interface<ReactiveElement>): V => {
+      // TODO: if we want to allow users to assert that the query will never
+      // return null, we need a new option and to throw here if the result
+      // is null.
       return (el.renderRoot?.querySelector(selector) ?? null) as V;
     };
-    if (typeof nameOrContext === 'object') {
+    if (cache) {
```

Comment:
consider structuring this like:

```ts
if (!cache) {
  // simple case, just return the doQuery getter
}
// harder case, with caching
```

I think it will read a bit better